edweiss-app/app/(app)/lectures/remotecontrol/quizToSlide.tsx

```diff
@@ -0,0 +1,160 @@
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import For from '@/components/core/For';
+import RouteHeader from '@/components/core/header/RouteHeader';
+import Icon from '@/components/core/Icon';
+import TActivityIndicator from '@/components/core/TActivityIndicator';
+import TText from '@/components/core/TText';
+import { QuizBroadcastModal } from '@/components/lectures/remotecontrol/modal';
+import { CollectionOf } from '@/config/firebase';
+import t from '@/config/i18config';
+import { ApplicationRoute } from '@/constants/Component';
+import { useDynamicDocs, usePrefetchedDynamicDoc } from '@/hooks/firebase/firestore';
+import LectureDisplay from '@/model/lectures/lectureDoc';
+import { LectureQuizzes } from '@/model/quizzes';
+import { BottomSheetModal } from '@gorhom/bottom-sheet';
+import { useLocalSearchParams } from 'expo-router';
+import React, { useRef, useState } from 'react';
+import { Vibration } from 'react-native';
+
+// Types
+type Lecture = LectureDisplay.Lecture;
+type Quiz = LectureDisplay.QuizLectureEvent;
+
+// ------------------------------------------------------------
+// ----------------    Quiz To Slide Screen     ---------------
+// ------------------------------------------------------------
+
+const QuizToSlideScreen: ApplicationRoute = () => {
+    // Argument Processing
+    const { courseNameString, lectureIdString } = useLocalSearchParams();
+    const courseName = courseNameString as string;
+    const lectureId = lectureIdString as string;
+
+
+
+    // Modal References
+    const modalRefQuizBroadcast = useRef<BottomSheetModal>(null);
+
+    //Hooks
+    const [selID, setSelID] = useState("");
+    const [selQuiz, setSelQuiz] = useState<LectureQuizzes.LectureQuiz>();
+    const [broadcasted, setBroadcasted] = useState(""); // Store the currently broadcasted quiz 
+
+
+
+    // Dynamic Document
+    const [lectureDoc] = usePrefetchedDynamicDoc(CollectionOf<Lecture>(`courses/${courseName}/lectures`), lectureId, undefined);
+    const quizDoc = useDynamicDocs(CollectionOf<Quiz>(`courses/${courseName}/lectures/${lectureId}/lectureEvents`));
+
+    // Wait for lectureDoc to be available
+    if (!lectureDoc) return <TActivityIndicator size={40} testID='quiz-slide-activity-indicator' />;
+    quizDoc?.sort((a, b) => a.data.pageNumber - b.data.pageNumber);
+    const quizDocPending = quizDoc?.filter(quiz => !quiz.data.done) || [];
+    const currentLecture = lectureDoc.data;
+
+    // Set the current broadcasted if the user just 
+    // comes back from another screen
+    if (broadcasted === "" && currentLecture.event && currentLecture.event.type === "quiz") {
```

Comment:
I think !broadcasted could have also worked here instead of broadcasted === "" as it is cleaner and more readable

Reply:
And I uselly do not use 'if's by themselves either as I do not exactly know in these scenarios if they are being called many times or just once. I usually opt to using useEffects but maybe I am just not all that well informed on the subject